src/ts/view/RechargeView.ts

```diff
@@ -0,0 +1,72 @@
+import { VendingMachineInterface } from '../domain/VendingMachine';
+import { $ } from '../utils';
+
+export interface RechargeViewInterface {
+  $rechargeForm: HTMLFormElement;
+  $currentHoldingMoney: HTMLSpanElement;
+  $rechargeInput: HTMLInputElement;
+  $coin500: HTMLSpanElement;
+  $coin100: HTMLSpanElement;
+  $coin50: HTMLSpanElement;
+  $coin10: HTMLSpanElement;
+  vendingMachine: VendingMachineInterface;
+
+  handleSubmit(event: SubmitEvent): void;
+  renderHoldingMoney(): void;
+  renderRecharge(): void;
+  renderCoinTable(): void;
+}
+
+export default class RechargeView implements RechargeViewInterface {
+  $rechargeForm: HTMLFormElement;
+  $currentHoldingMoney: HTMLSpanElement;
+  $rechargeInput: HTMLInputElement;
+  $coin500: HTMLSpanElement;
+  $coin100: HTMLSpanElement;
+  $coin50: HTMLSpanElement;
+  $coin10: HTMLSpanElement;
+  vendingMachine: VendingMachineInterface;
+
+  constructor(vendingMachine: VendingMachineInterface) {
+    this.$rechargeForm = <HTMLFormElement>$('.recharge-form');
+    this.$currentHoldingMoney = $('#current-holding-money');
+    this.$rechargeInput = <HTMLInputElement>$('#recharge-input', this.$rechargeForm);
+    this.$coin500 = <HTMLSpanElement>$('#coin-500');
+    this.$coin100 = <HTMLSpanElement>$('#coin-100');
+    this.$coin50 = <HTMLSpanElement>$('#coin-50');
+    this.$coin10 = <HTMLSpanElement>$('#coin-10');
+    this.vendingMachine = vendingMachine;
+
+    this.$rechargeForm.addEventListener('submit', this.handleSubmit);
+  }
+
+  renderCoinTable = () => {
+    this.$coin500.textContent = String(this.vendingMachine.getCoinByValue(500).count);
+    this.$coin100.textContent = String(this.vendingMachine.getCoinByValue(100).count);
+    this.$coin50.textContent = String(this.vendingMachine.getCoinByValue(50).count);
+    this.$coin10.textContent = String(this.vendingMachine.getCoinByValue(10).count);
+  };
+
+  renderRecharge = () => {
+    this.renderHoldingMoney();
+    this.renderCoinTable();
+    this.$rechargeInput.focus();
+  };
+
+  renderHoldingMoney = () => {
+    this.$currentHoldingMoney.textContent = String(this.vendingMachine.getHoldingMoney());
+  };
+
+  handleSubmit = (event: SubmitEvent) => {
+    event.preventDefault();
+    const moneyToRecharge = +this.$rechargeInput.value;
+
+    try {
+      this.vendingMachine.rechargeMoney(moneyToRecharge);
+      this.renderRecharge();
+      this.$rechargeInput.value = '';
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+}
```

Comment:
```suggestion

  handleSubmit = (event: SubmitEvent) => {
    event.preventDefault();
    const moneyToRecharge = +this.$rechargeInput.value;

    try {
      this.vendingMachine.rechargeMoney(moneyToRecharge);
      this.renderRecharge();
      this.$rechargeInput.value = '';
    } catch (error) {
      alert(error.message);
    }
  };

  renderRecharge = () => {
    this.renderHoldingMoney();
    this.renderCoinTable();
    this.$rechargeInput.focus();
  };

  renderHoldingMoney = () => {
    this.$currentHoldingMoney.textContent = String(this.vendingMachine.getHoldingMoney());
  };

  renderCoinTable = () => {
    this.$coin500.textContent = String(this.vendingMachine.getCoinByValue(500).count);
    this.$coin100.textContent = String(this.vendingMachine.getCoinByValue(100).count);
    this.$coin50.textContent = String(this.vendingMachine.getCoinByValue(50).count);
    this.$coin10.textContent = String(this.vendingMachine.getCoinByValue(10).count);
  };
```
코드도 결국 글이고 문장이기 때문에 위에서 아래로 읽는게 편합니다. 코드를 읽다가 위로 올라가게 된다면 흐름이 깨지고 불편하기 때문에 배치도 신경쓰면 좋을것 같아요. 역할에 따라서 나누고 읽는 흐름의 순서를 고려해서 배치하는거죠.

Reply:
넵! 이 부분은 앞에서 리팩토링하다가 인지하여 전체적으로 수정하였습니다. 예시로 들어주신 것처럼, 신문기사를 읽듯이 위에서 아래로 자연스럽게 읽히도록 수정해보았습니다.
그리고 public 메서드를 쉽게 찾기 위해 constructor 메서드 바로 밑에 두는 것으로 수정해보았습니다!

Reply:
보통 private이 public보다 먼저 나오는걸로 알고 있습니다